packages/db/src/search/StatsSearch.ts

```diff
@@ -0,0 +1,298 @@
+    const dateto = this.queryStringParameters.timestamp__to ?
+      new Date(timestampTo).toISOString() : new Date().toISOString();
+    const datefrom = this.queryStringParameters.timestamp__from ?
+      new Date(timestampFrom).toISOString() : '1970-01-01T12:00:00+00:00';
+    return {
+      errors: {
+        dateFrom: datefrom,
+        dateTo: dateto,
+        value: Number(result.count_errors),
+        aggregation: 'count',
+        unit: 'error',
+      },
+      collections: {
+        dateFrom: datefrom,
+        dateTo: dateto,
+        value: Number(result.count_collections),
+        aggregation: 'count',
+        unit: 'collection',
+      },
+      processingTime: {
+        dateFrom: datefrom,
+        dateTo: dateto,
+        value: Number(result.avg_processing_time),
+        aggregation: 'average',
+        unit: 'second',
+      },
+      granules: {
+        dateFrom: datefrom,
+        dateTo: dateto,
+        value: Number(result.count_granules),
+        aggregation: 'count',
+        unit: 'granule',
+      },
+    };
+  }
+
+  /** Queries postgres for a summary of statistics around the granules in the system
+   *
+   * @param {Knex} sendKnex - the knex client to be used
+   * @returns {Promise<SummaryResult>} the postgres aggregations based on query
+   */
+  public async summary(sendknex: Knex): Promise<SummaryResult> {
+    const knex = sendknex ?? await getKnexClient();
+    const aggregateQuery:Knex.QueryBuilder = knex(`${TableNames.granules}`);
+    if (this.queryStringParameters.timestamp__from) {
+      aggregateQuery.where(`${TableNames.granules}.updated_at`, '>=', new Date(Number.parseInt(this.queryStringParameters.timestamp__from as string, 10)));
+    }
+    if (this.queryStringParameters.timestamp__to) {
+      aggregateQuery.where(`${TableNames.granules}.updated_at`, '<=', new Date(Number.parseInt(this.queryStringParameters.timestamp__to as string, 10)));
+    }
+    aggregateQuery.select(
+      knex.raw(`COUNT(CASE WHEN ${TableNames.granules}.error ->> 'Error' != '{}' THEN 1 END) AS count_errors`),
+      knex.raw(`COUNT(${TableNames.granules}.cumulus_id) AS count_granules`),
+      knex.raw(`AVG(${TableNames.granules}.duration) AS avg_processing_time`),
+      knex.raw(`COUNT(DISTINCT ${TableNames.granules}.collection_cumulus_id) AS count_collections`)
+    );
+    const aggregateQueryRes: TotalSummary[] = await aggregateQuery;
+    return this.formatSummaryResult(aggregateQueryRes[0]);
+  }
+
+  /** Performs joins on the provider and/or collection table if neccessary
+   *
+   * @param {Knex} knex - the knex client to be used
+   * @returns {Knex.QueryBuilder} the knex query of a joined table or not based on queryStringParams
+   */
+  private providerAndCollectionIdBuilder(knex: Knex): Knex.QueryBuilder {
+    let aggregateQuery;
+    this.queryStringParameters.field = this.queryStringParameters.field ? this.queryStringParameters.field : 'status';
```

Comment:
Avoid changing this.queryStringParameters which holds original input information, use local variable.